### ISSUE TITLE
Перестали проигрываться сериалы на mi box 3

### ISSUE THREAD

**Issue by nsinko** [vmasalov/seasonvar.bundle#15]
Доброй день. 

Прошу помощи, готов предоставить все требуемое для решения. Суть проблемы в сабже. 

Окружение - плех на сайнолоджи, установленный плагин, премиум аккаунт. Смотрю через mi box 3 (нативное приложение plex из google play). 

Несколько дней назад все работало нормально. А вчера перестало. Вначале плагин на mi box вообще не открывался (вернее открывался со словами что то типа "папка пустая"). После перезапуска плекса (остановил и запустил заново) все открывается, вижу сериалы, но при попытке проиграть пишет "воспроизведение не возможно". В браузере на компе при этом все воспроизводится.

Готов посмотреть логи или сделать что то еще, скажите что :) Буду очень благодарен за помощь.

**Comment by vmasalov** [comment]
Что бы активировать логи нужно:
1. Открываем страницу плех сервера http://**_IP_ADDRESS_PLEX_SERVER_**:32400/web/index.html
> _IP_ADDRESS_PLEX_SERVER_ замени на IP типо 192.168.1.1 
2. Открываем настройки Settings -> Server -> General. 
Активируй `Show Advanced` и установи галочки на следующие пункты:
```
[x] Enable Plex Media Server debug logging
[x] Enable Plex Media Server verbose logging
[x] Allow Plex Media Server tokens in logs
```
Дальше самое интересное... :)

Можно скачать логи с  Help -> Download Logs, но там не всегда все есть что нужно
> Это последний пункт меню Settings -> Server

Я кручу plex на raspberryPi и туда настроил SSH доступ. Если есть возможность доступиться на файловую систему то логи можно найти `/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Logs/PMS Plugin Logs` 
> Путь может координально отличаться.

Обычно логи к плагину находятся в `com.plexapp.plugins.seasonvar.log` файле. Поэтому можно попробовать поиск этого файла.

В крайнем случае если нет возможности доступиться до логов вообще никак... то тогда самое оптимальное это выводить сообщение на экран. 

```
display_message(title="HELLO", message="WORLD")
```

Очень рекомендую посмотреть манул ст. 92 (Debugging) так как может понадобится добавлять код отладки.

[https://us.v-cdn.net/6025034/uploads/editor/cm/cojaep28kmq3.pdf](PlexPluginFramworkAPI.pdf)

Happy debugging :)

**Comment by vmasalov** [comment]
Кстати может быть проблема медленного интернета. У меня как-то такое было на телевизоре когда он на wifi был, подключил к кабелю - больше не видел этой проблемы

**Comment by nsinko** [comment]
Добрый день. Большое спасибо за ответ. Но боюсь Вы меня переоценили - дебаг для меня пожалуй перебор :). Но все же я попробую. Код на питоне уже совсем не про меня - слишком давно не брал я в руки что то такое.

Расширенное логгирование сервера плекса я включу, тут все понятно. Логи так же скачаю. Но как я Вас понял, мне нужно дополнительно добраться до логов на клиентском устройстве. Верно? Но их сбор как мне кажется надо включить на самом устройстве.

Доступа к файловой системе у меня нет, но попробую через файловый менеджер на самом устройстве поискать.

**Comment by nsinko** [comment]
Впрочем я наверное Вас не правильно понял - речь о серверных логах, но нужен лог именно плагина. Сервер у меня на сайнолоджи и доступ к файловой системе есть. Лог я нашел. Вот только понять его не могу. Выглядит все ок, кроме вот этой строки:

2018-05-15 18:49:32,832 (-baf4490) :  WARNING (objectkit:208) - Media part has no streams - attempting to synthesize

Файл прикреплю. Если не сложно - посмотрите плиз. Он короткий.
[com.plexapp.plugins.seasonvar.log](https://github.com/vmasalov/seasonvar.bundle/files/2005815/com.plexapp.plugins.seasonvar.log)

**Comment by nsinko** [comment]
Для интереса попробовал с айпада запустить. Ошибка другая и лог другой. То же приложил на всякий случай.
[com.plexapp.plugins.seasonvar-2.log](https://github.com/vmasalov/seasonvar.bundle/files/2005864/com.plexapp.plugins.seasonvar-2.log)

**Comment by vmasalov** [comment]
Все нормально, у меня +/- такой же лог. И работает в браузере. 

Попробовал открыть на плашете, валит ошибку тоже. Тут походу что-то другое. Посмотрю в чем проблема.

**Comment by nsinko** [comment]
Да, первый лог выглядит нормальным, но проигрывать то не хочет! Я сделал предположение, что возможно плагин отработал нормально, а дальше типа передал стрим самому плексу и уже там что то не срабатывает. Посмотрел время окончания лога плагина - 18:49:34 и проверил Plex Media Server.log с этого времени. Если честно толком ничего не понял, не понравилось только вот эта строка:

May 15, 2018 18:50:02.642 [0xf17ffb70] WARN - LongPoll: Got error, closing.

Не посмотрите лог?
[Plex Media Server.log](https://github.com/vmasalov/seasonvar.bundle/files/2008947/Plex.Media.Server.log)

P.S. Так же проверил остальные логи. В это время были записи еще только в com.plexapp.system.log, но там все совсем чисто.

**Comment by vmasalov** [comment]
А вообще видео запускается не с канала? Т.е. если файл локально добавлен в медиа?
Потому что судя по форумам, народ жалуется на ту же проблему и это не связано с плагином.

**Comment by nsinko** [comment]
В том то и дело, что все видео в плексе (фильмы, сериалы) в различных форматах проигрываются нормально. Особенностью может являться разве что я везде настроил оригинальное качество передачи, чтобы не грузить проц на перекодировке. Но так было с самого начала.

Завтра попоробую поиграться с настройками на клиентском ми бокс, но что то сомневаюсь, что поможет.

**Comment by ReinVitter** [comment]
Добрый день.
Не проигрываются сериалы. Сервак на винде, заходит в сериал, выбираю перевод - пишет "подключаемый модуль не отвечает".
Помогите плиз
[Plex Media Server Logs_2018-06-17_00-49-06.zip](https://github.com/vmasalov/seasonvar.bundle/files/2108530/Plex.Media.Server.Logs_2018-06-17_00-49-06.zip)
[com.plexapp.plugins.seasonvar.log](https://github.com/vmasalov/seasonvar.bundle/files/2108532/com.plexapp.plugins.seasonvar.log)

**Comment by vmasalov** [comment]
Да я вижу ошибки которую кидает плагин.

Возникает 2а вопроса
1. У вас плагин на русском или англ языке? Если на русском - можете переключить на анг. и посмотреть будет ли работать или нет.

2. Только определенный сериал не работает или вообще ничего?

**Comment by ReinVitter** [comment]
1. На русском. Переключаю настройки сервера на английский - "This plugin is not responding."
2. Штук 5 проверил - не работает.

**Comment by vmasalov** [comment]
О как интерестно. А можешь переключить на англ и скинуть мне лог. Очень странно, потому что у народа неполучается переключить язык и обычно работает англ. версия...

**Comment by ReinVitter** [comment]
[com.plexapp.plugins.seasonvar.log](https://github.com/vmasalov/seasonvar.bundle/files/2116455/com.plexapp.plugins.seasonvar.log)
Получите... 
А что можно мне сделать? Переустановить плекс на английском? Или что можно попробовать?

**Comment by vmasalov** [comment]
Еще актульно? А то я ту выпал на пару месяцев из проекта

**Comment by ReinVitter** [comment]
актуально, но давно не проверял. сегодня куплю премиум и проверю

пт, 21 сент. 2018, 7:47 Vladimir M <notifications@github.com>:

**Comment by ReinVitter** [comment]
не работает. Список сериалов показывает, а когда нажимаю воспроизвести -
пишет проверьте доступность файла.

пт, 21 сент. 2018 г. в 7:49, Rein Vitter <reinvitter@gmail.com>:

**Comment by vmasalov** [comment]
Т.е получается, список получает а при попытке проиграть - складывается... А это все сериалы или только определенный?

**Comment by ReinVitter** [comment]
Все. При том на другой комп, на свежеустановленную семерку поставил чистый
plex - та же история.

пн, 29 окт. 2018 г. в 22:40, Vladimir M <notifications@github.com>:

**Comment by vmasalov** [comment]
Ок, а если попробовать через браузерную версию?... типо 127.0.0.1:32400/web/index.html работает?
